### ISSUE TITLE
Make it Possible to Add Child Folders as Root Folders

### ISSUE THREAD

**Issue by craighooghiem** [atom/tree-view#404]
Now that we can add root folders to the tree-view (https://github.com/atom/tree-view/issues/63) it would be great if we could make a small adjustment to make it function even more like alternative editors do.

Currently I can add additional root folders to a project to allow for a more organized tree view, but I cannot add a root folder if it is a child of an existing root folder. 

**Use Case:** Working on a project with a library or vendor folder that goes multiple folders deep. I would like to be able to organize the tree view better to make it easier to navigate these folders.

![screen shot 2015-03-12 at 7 37 52 am](https://cloud.githubusercontent.com/assets/1476306/6617072/1d1aab1a-c88b-11e4-9fe7-cd035d09cb7b.png)

![screen shot 2015-03-12 at 7 38 17 am](https://cloud.githubusercontent.com/assets/1476306/6617074/22868074-c88b-11e4-9852-dbf946c8d8a7.png)

**Comment by fpupor** [comment]
I think that this function is really necessary, I use a lot in other editors

**Comment by timodwhit** [comment]
:+1:

**Comment by mathiasvr** [comment]
This can be achieved by simply removing the `if directory.contains(projectPath)` condition at [line 211 in project.coffee](https://github.com/atom/atom/blob/master/src/project.coffee#L211). But maybe it should be indicated when child folders are added like this.

**Comment by kebertx** [comment]
I agree, wholeheartedly!

The way the comment above [project.coffee Line 193](https://github.com/atom/atom/blob/master/src/project.coffee#L193) is written makes me think that not duplicating entries in `project.rootDirectories` was an afterthought, and the main idea is to prevent users from doing exactly this:
```
      # Apparently a Directory does not believe it can contain itself, so we
      # must also check whether the paths match.
      return if directory.contains(projectPath) or directory.getPath() is projectPath
```
It's a totally valid use case though. I mean, I'd like the flexibility, anyway.

**Comment by marjakm** [comment]
Would like to have this feature

**Comment by maxbrunsfeld** [comment]
Hey everyone, sorry for the wait. This will be fixed in Atom 1.0.8.
